freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -22,40 +24,46 @@
 
 @Service
 public class UmsUsersStateProvider {
     private static final Logger LOGGER = LoggerFactory.getLogger(UmsUsersStateProvider.class);
 
-    private static ThreadLocal<String> requestIdThreadLocal = new ThreadLocal<>();
-
     @Inject
-    private GrpcUmsClient umsClient;
+    private GrpcUmsClient grpcUmsClient;
 
     public Map<String, UsersState> getEnvToUmsUsersStateMap(String accountId, String actorCrn, Set<String> environmentCrns,
-                                                            Set<String> userCrns, Set<String> machineUserCrns) {
+                                                            Set<String> userCrns, Set<String> machineUserCrns, Optional<String> requestId) {
         try {
-            Optional<String> requestIdOptional = Optional.ofNullable(MDCBuilder.getMdcContextMap().get(LoggerContextKey.REQUEST_ID.toString()));
-            if (!requestIdOptional.isPresent()) {
-                requestIdOptional = Optional.of(UUID.randomUUID().toString());
-                LOGGER.debug("No requestId found. Setting request id to new UUID [{}]", requestIdOptional);
+            final Optional<String> requestIdNewIfEmpty;
+            if (!requestId.isPresent()) {
+                requestIdNewIfEmpty = Optional.of(UUID.randomUUID().toString());
+                LOGGER.debug("Environment Crns: {}, no requestId found. Setting request id to new UUID [{}]", environmentCrns, requestIdNewIfEmpty);
+            } else {
+                requestIdNewIfEmpty = requestId;
             }
-            requestIdThreadLocal.set(requestIdOptional.get());
-            LOGGER.debug("Getting UMS state for environments {} with requestId {}", environmentCrns, requestIdOptional);
+            LOGGER.debug("Getting UMS state for environments {} with requestId {}", environmentCrns, requestIdNewIfEmpty);
 
             Map<String, UsersState> envUsersStateMap = new HashMap<>();
 
-            List<User> users = userCrns.isEmpty() ? umsClient.listAllUsers(actorCrn, accountId, requestIdOptional)
-                    : umsClient.listUsers(actorCrn, accountId, List.copyOf(userCrns), requestIdOptional);
-            List<MachineUser> machineUsers = machineUserCrns.isEmpty() ? umsClient.listAllMachineUsers(actorCrn, accountId, requestIdOptional)
-                    : umsClient.listMachineUsers(actorCrn, accountId, List.copyOf(machineUserCrns), requestIdOptional);
+            List<User> users = userCrns.isEmpty() ? grpcUmsClient.listAllUsers(actorCrn, accountId, requestIdNewIfEmpty)
+                : grpcUmsClient.listUsers(actorCrn, accountId, List.copyOf(userCrns), requestIdNewIfEmpty);
+
+            List<MachineUser> machineUsers = machineUserCrns.isEmpty() ? grpcUmsClient.listAllMachineUsers(actorCrn, accountId, requestIdNewIfEmpty)
+                : grpcUmsClient.listMachineUsers(actorCrn, accountId, List.copyOf(machineUserCrns), requestIdNewIfEmpty);
+
+
+//            // get users hashed password
```

Comment:
This commented out code looks like it is irrelevant due to your changes to `handleUser`. You should remove it